### PR TITLE
upload artifacts in github actions for debugging

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

CI: upload server logs as artifacts in github actions, so we can debug better. We need this for issues that are hard to reproduce, like when they only happen sometimes (flaky tests).

## GUI diff

No difference.

- [ ] **DONE**

## Documentation
- No documentation needed

- [ ] **DONE**

## Test coverage
- No tests

- [ ] **DONE**

## Links

N/A

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: .github/workflows/acceptance_tests_common.yml [hunk 1]
```diff
@@ -84,4 +87,13 @@ jobs:
         run: ./testsuite/podman_runner/18_split_secondary_p_tests.sh
       - name: run_secondary_tests
         run: ./testsuite/podman_runner/${{ inputs.secondary_tests }} 
+      - name: get_server_logs
+        if: ${{ failure() }}
```
Comment (srbarrios) [inline .github/workflows/acceptance_tests_common.yml:91]: If it doesn't cost resources... I would save the logs always, not only when failing.

File: testsuite/podman_runner/19_get_server_logs.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	echo "Usage: ${0} server_id"
+	echo "server_id is used for creating a unique folder"
+	exit -1
+fi
+
+src_dir=$(cd $(dirname "$0")/../.. && pwd -P)
+server_id=${1}
+rm -rfv /tmp/test-all-in-one/server-logs/${server_id}
+mkdir -p /tmp/test-all-in-one/server-logs/${server_id}
+sudo -i podman exec uyuni-server-all-in-one-test bash -c "tar jcvf /tmp/server-logs/${server_id}/var.log.rhn.tar.bz2 /var/log/rhn/"
```
Comment (srbarrios) [inline testsuite/podman_runner/19_get_server_logs.sh:15]: We don't need those, are already included in the supportconfig.

File: testsuite/podman_runner/19_get_server_logs.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+sudo -i podman exec uyuni-server-all-in-one-test bash -c "supportconfig -R /tmp/server-logs/${server_id} && chmod 644 /tmp/server-logs/${server_id}/scc_uyuni-server-all-in-one-test*.txz*"
```
Comment (srbarrios) [inline testsuite/podman_runner/19_get_server_logs.sh:16]: Be aware that for the supportconfig to collect proper logs from SUMA, we need to have installed the plugins for SUMA

File: .github/workflows/acceptance_tests_common.yml (hunk 1 above)
Comment (jordimassaguerpla) [reply .github/workflows/acceptance_tests_common.yml:91]: It takes time ... 5 minutes aprox. There is a limit on disc space too, but I do not think  we have a problem there. I rather prefer to not take these extra 5 minutes than to have the logs when it succeeds.

File: testsuite/podman_runner/19_get_server_logs.sh (hunk 3 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/19_get_server_logs.sh:16]: I checked the logs and I saw a spacewalk-debug folder. I assume it is installed, given we use the pattern to install the already released packages.

File: testsuite/podman_runner/19_get_server_logs.sh (hunk 2 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/19_get_server_logs.sh]: ok. I have removed it .

File: .github/workflows/acceptance_tests_common.yml (hunk 1 above)
Comment (jordimassaguerpla) [reply .github/workflows/acceptance_tests_common.yml:91]: What do you think?

Comment (srbarrios) [reply .github/workflows/acceptance_tests_common.yml:91]: Let's skip it, and if we ever find a good reason to have the logs in passed tests, we can think about adding them in some conditional way (using some label in the GitHub PR).

Comment (jordimassaguerpla) [reply .github/workflows/acceptance_tests_common.yml:91]: We agreed on slack that it is ok to skip.

**Pull request merged**